app/main.py

```diff
@@ -2,5 +2,17 @@
 
 
 def cache(func: Callable) -> Callable:
-    # Write your code here
-    pass
+    cached_results = {}
+
+    def wrapper(*args, **kwargs) -> int:
```

Comment:
Add a decorator from `functools` for `wrapper` to preserve the wrapped function name and docstring.